src/main/java/mate/academy/dao/impl/AbstractDao.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public abstract class AbstractDao {
+    protected final SessionFactory factory;
+
+    public AbstractDao(SessionFactory factory) {
+        this.factory = factory;
+    }
+
+    protected <T> T create(T entity) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
```

Comment:
You are using `HibernateUtil.getSessionFactory().openSession()` to open a session, but you should use the `factory` field that is passed to the constructor of `AbstractDao` instead. This will make your code more flexible and easier to test because it won't be tightly coupled to the HibernateUtil class.